libbeat/cmd/instance/metrics.go

```diff
@@ -36,3 +55,38 @@ func reportInfo(_ monitoring.Mode, V monitoring.Visitor) {
 	uptime := int64(delta.Seconds())*1000 + int64(delta.Nanoseconds()/1000000)
 	monitoring.ReportInt(V, "uptime.ms", uptime)
 }
+
+func reportCPU(_ monitoring.Mode, V monitoring.Visitor) {
+	V.OnRegistryStart()
+	defer V.OnRegistryFinished()
+
+	cpuUsage, normalizedCPU := getCPUUsage()
+	monitoring.ReportFloat(V, "usage", cpuUsage)
+	monitoring.ReportFloat(V, "usage.normalized", normalizedCPU)
+}
+
+func getCPUUsage() (float64, float64) {
+	pid := os.Getpid()
+
+	sample := struct {
+		time      time.Time
+		procTimes sigar.ProcTime
+	}{
+		time.Now(),
+		sigar.ProcTime{},
+	}
+
+	if err := sample.procTimes.Get(pid); err != nil {
+		return 0, 0
+	}
+
+	dTime := sample.time.Sub(lastSample.time)
+	dMilli := dTime / time.Millisecond
+	dCPU := int64(sample.procTimes.Total - lastSample.procTimes.Total)
+
+	usage := float64(dCPU) / float64(dMilli)
+	normalized := usage / float64(numCores)
```

Comment:
I think I might be missing some context here concerning the nature of the recording.
This will give us a rate of cpu usage?

Reply:
Yes, it gives the total CPU usage of a beat between the last two samples.

I used the method from `metricbeat/modules/system/process/helper`. I thought about extracting the whole `Process` and `ProcStats` and moving it `libbeat`. However, it seemed unnecessary to move such a big part of `metricbeat` into `libbeat` for only this minor calculation. But I am open to reconsider moving it.

Reply:
All fine then! thanks @kvch !